### PR TITLE
[Dashboard] made loading screen between pages from dashboard page

### LINKED ISSUE #105: [Dashboard] Animations between pages
### Objectives
Create animations (loading screen) for transitions between login page, dashboard, journal, todo, and calendar

### Acceptance Criteria
- [ ] Loading screen between login page and dashboard
- [ ] Loading screen between dashboard and journal (both ways)
- [ ] Loading screen between dashboard and todo (both ways)
- [ ] Loading screen between dashboard and calendar (both ways)
- [ ] Loading screen between calendar, todo and journal (three ways)
- [ ] 404 page not found page if page was not found

### Assignees
- @sanjithdevineni

### PR DESCRIPTION
**Pull request by sanjithdevineni**
# Objective
Refer to the corresponding GitHub issue #105.

The purpose of this PR is to implement the feature of a loading screen between pages. A loading screen will pop up when going to dashboard, journal, todo or calendar from the dashboard

image

## Solution
- Add new div called 'loading-screen'
- Add new event listener for all links that displays the div

### Assignees
- @sanjithdevineni

### CODE REVIEW

File: assets/scripts/main.js [hunk 1]
```diff
@@ -164,6 +164,37 @@ function updateMoodChart(mood) {
...
+    iframe.style.display = 'none';
+    iframe.src = targetUrl;
+    document.body.appendChild(iframe);
+
+    iframe.onload = () => {
+      clearTimeout(loadingTimeout); // Clear the timeout if the page loads quickly
+      window.location.href = targetUrl; // Proceed to the target URL
+  };
+      
+      // // Redirect after a short delay (e.g., 3 seconds)
```
Comment (adironene) [inline assets/scripts/main.js]: take this out if we are not using it in production

**Review by adironene — CHANGES_REQUESTED** [review]
just take out the comment. apart from that, looks great. great job testing with slow internet setting

**Review by kabirsachdev7 — APPROVED** [review]
approved, good stuff on changing when the loading happens. checked `network` and confirmed that the loading occurs when on slower network.

**Comment by kabirsachdev7** [comment]
pls resolve conflicts

**Pull request merged**